src/SFA.DAS.ApprenticeFeedback.Application.UnitTests/Commands/WhenHandlingCreateApprenticeFeedbackCommand.cs

```diff
@@ -0,0 +1,70 @@
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApprenticeFeedback.Application.UnitTests.Commands
+{
+    public class WhenHandlingCreateApprenticeFeedbackCommand
+    {
+        [Test, MoqAutoData]
+        public async Task And_GetFeedbackTargetIsNull_Then_ThrowException
+            (CreateApprenticeFeedbackCommand command,
+            [Frozen] Mock<IApprenticeFeedbackRepository> mockApprenticeFeedbackRepository,
+            CreateApprenticeFeedbackHandler handler)
+        {
+            mockApprenticeFeedbackRepository.Setup(s => s.GetApprenticeFeedbackTargetById(command.ApprenticeId)).Throws(new Exception());
+
+            Func<Task> result = async () => await handler.Handle(command, CancellationToken.None);
+
+            await result.Should().ThrowAsync<Exception>();
+        }
+
+        [Test, MoqAutoData]
+        public async Task And_GetFeedbackTargetIsValid_And_GetAttributesIsNotValid_Then_ThrowException
+            (CreateApprenticeFeedbackCommand command,
+            [Frozen] Mock<IApprenticeFeedbackRepository> mockApprenticeFeedbackRepository,
+            CreateApprenticeFeedbackHandler handler,
+            Domain.Entities.ApprenticeFeedbackTarget apprenticeFeedbackTarget)
+        {
+            mockApprenticeFeedbackRepository.Setup(s => s.GetApprenticeFeedbackTargetById(command.ApprenticeId)).ReturnsAsync(apprenticeFeedbackTarget);
+            mockApprenticeFeedbackRepository.Setup(s => s.GetAttributes());
+
+            Func<Task> result = async () => await handler.Handle(command, CancellationToken.None);
+
+            await result.Should().ThrowAsync<Exception>();
+            
+        }
+
+        [Test, RecursiveMoqAutoData]
+        public async Task And_GetFeedbackTargetIsValid_And_GetAttributesIsValid_Then_CreateNewFeedbackRecord_And_ReturnResponse
+            (CreateApprenticeFeedbackCommand command,
+            [Frozen] Mock<IApprenticeFeedbackRepository> apprenticeFeedbackRepository,
+            Domain.Entities.ApprenticeFeedbackTarget apprenticeFeedbackTarget,
+            ApprenticeFeedbackResult apprenticeFeedbackResult,
+            CreateApprenticeFeedbackHandler handler,
+            IEnumerable<Domain.Entities.Attribute> attributes)
+        {
+            command.FeedbackAttributes = attributes.Select(a => new Domain.Models.FeedbackAttribute { Id = a.AttributeId, Name = a.AttributeName, Status = FeedbackAttributeStatus.Agree }).ToList();
+            apprenticeFeedbackRepository.Setup(s => s.GetApprenticeFeedbackTargetById(command.ApprenticeId)).ReturnsAsync(apprenticeFeedbackTarget);
+            apprenticeFeedbackRepository.Setup(s => s.GetAttributes()).ReturnsAsync(attributes);
+            apprenticeFeedbackRepository.Setup(s => s.CreateApprenticeFeedbackResult(apprenticeFeedbackResult));
+
+            var result = await handler.Handle(command, CancellationToken.None);
+
+
+            result.Should().BeOfType<CreateApprenticeFeedbackResponse>();
```

Comment:
We can verify the repository method here, and make sure the Create method takes in the feedback object as created

apprenticeFeedbackRepository.Verify(s => s.CreateApprenticeFeedbackResult(It.Is<ApprenticeFeedbackResult>( a => ....

Reply:
I have added this line of code in `apprenticeFeedbackRepository.Verify(s => s.CreateApprenticeFeedbackResult(It.IsAny<ApprenticeFeedbackResult>()), Times.Once());`